src/main/java/com/example/demo/services/BookServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.example.demo.services;
+
+import com.example.demo.model.Book;
+import com.example.demo.repository.BookRepository;
+import com.example.demo.dto.BookMainInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class BookServiceImpl implements BookService {
```

Comment:
Tutaj jest straszne zamieszanie. Najpierw powinny być wszystkie publiczne metody, żeby nie szukać, potem prywatne. Wydaje mi się też, że za bardzo to jest rozbite i się robi trochę taki spaghetti code

Reply:
według clean code, metody powinny być w takiej kolejności, że jak wywołujesz jedną a w niej drugą, to ta druga powinna się znajdować zaraz pod nią, nie trzba wtedy latać po kodzie góra dół, żeby zobaczyć co wywołują metody publiczne. Też jest tam zalecenie, że dobrze jest pisać małe metody, bo łatwiej je ogarnąć. 
Przeczytałam clean code, więc korzystam z tamtych rad i wydają mi się sensowne

Reply:
Pytanie co to jest z definicji mała metoda:
private Stream<Book> getBooksStream() { return findAll().stream(); }
Samo stream() jest już metodą. To wszystko jest wrapperem, który dokłada tylko linijki kodu. Zgadzam się, że lepiej mieć małe metody, ale też trzeba zachować tutaj pewien rozsądek.

Clean code ma trochę fajnych teorii i wszystko fajnie działa póki mamy prosty kod. Teraz wyobraźmy sobie sytuacje, że z którejś naszej metody korzystają 3 inne, a z kolejnej 5 innych. To gdzie tą metodę wstawić? Pod pierwszym wywołaniem, pod ostatnim? W przypadku gdy mamy publiczne metody najpierw, programista od razu zobaczy co dany serwis udostępnia. Być może nawet nie musi się zagłębiać co jest dalej.

Mi osobiście ciężko było zacząć w ogóle sprawdzać ten serwis, bo nie wiedziałem gdzie zacząć, a klasa nie ma nawet 100 lini, a co będzie przy 1000? Ponieważ to moje osobiste odczucie, zrób jak uważasz :)

Reply:
metody które zwracają streamy powinny zostać spokojnie usunięte. Lambdy traktuj jako jedno, one są tak fajnie skonstruowane że podział na metody jedynie zamiesza.

Reply:
kilka metod wywaliłam, ale zostałam przy kolejności

Reply:
super. dla mnie to jak najbardziej odpowiada ;)